### PR TITLE
Start on breakage detection for structs

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: tools/Google.Cloud.Tools.VersionCompat/Detectors/Struct.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{
+    internal class Struct
+    {
+        public static IEnumerable<Diff> Diffs(TypeDefinition o, TypeDefinition n)
+        {
+            var strct = new Struct(o, n);
+            var classAndInterface = new ClassStructInterface(o, n);
+            // TODO: Check (instance) constructors
+            return Enumerable.Empty<Diff>()
+                //.Concat(strct.SealedAbstractStatic())
```
Comment (jskeet) [inline tools/Google.Cloud.Tools.VersionCompat/Detectors/Struct.cs]: Remove comment?

File: tools/Google.Cloud.Tools.VersionCompat/Detectors/TopLevel.cs [hunk 2]
```diff
@@ -48,10 +48,10 @@ public static IEnumerable<Diff> Diffs(IReadOnlyList<TypeDefinition> olderTypes,
                         switch (oType)
                         {
                             case TypeType.Class: typeDiffs = Class.Diffs(o, n); break;
+                            case TypeType.Struct: typeDiffs = Struct.Diffs(o, n);break;
```
Comment (jskeet) [inline tools/Google.Cloud.Tools.VersionCompat/Detectors/TopLevel.cs]: Nit: space before break

File: tools/Google.Cloud.Tools.VersionCompat.Tests/StructTest.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    namespace InterfaceRemoved.A { public struct S : I1 { } }
+    namespace InterfaceRemoved.B { public struct S { } }
+
+    namespace NonExportedInterfaceAdded.A { public struct S { } }
+    namespace NonExportedInterfaceAdded.B { public struct S : I2 { } }
+
+    namespace NonExportedInterfaceRemoved.A { public struct S : I2 { } }
+    namespace NonExportedInterfaceRemoved.B { public struct S { } }
+
+    namespace GenericConstraintAdded1.A { public struct S<A> { } }
```
Comment (jskeet) [inline tools/Google.Cloud.Tools.VersionCompat.Tests/StructTest.cs]: Should we have tests for `where A : unmanaged`? New in C# 7 IIRC. Can't see us using it, admittedly.

File: tools/Google.Cloud.Tools.VersionCompat.Tests/StructTest.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    namespace GenericConstraintChanged1.A { public struct S<A> where A : IList<A> { } }
+    namespace GenericConstraintChanged1.B { public struct S<A> where A : IEquatable<A> { } }
+
+    namespace GenericConstraintChanged2.A { public struct S<A, B> where A : IList<A> { } }
+    namespace GenericConstraintChanged2.B { public struct S<A, B> where A : IList<B> { } }
+
+    namespace GenericConstraintChanged3.A { public struct S<A, B> where A : IList<A> where B : IList<A> { } }
+    namespace GenericConstraintChanged3.B { public struct S<A, B> where A : IList<B> where B : IList<A> { } }
+}
+
```
Comment (jskeet) [inline tools/Google.Cloud.Tools.VersionCompat.Tests/StructTest.cs]: Presumably we'll have tests for adding methods etc? (Is there any reason for starting with generic constraints?)

File: tools/Google.Cloud.Tools.VersionCompat.Tests/StructTest.cs (hunk 3 above)
Comment (chrisdunelm) [reply tools/Google.Cloud.Tools.VersionCompat.Tests/StructTest.cs]: I guess so! TODO added in `ClassStructInterface.cs`.

File: tools/Google.Cloud.Tools.VersionCompat.Tests/StructTest.cs (hunk 4 above)
Comment (chrisdunelm) [reply tools/Google.Cloud.Tools.VersionCompat.Tests/StructTest.cs]: Yes, these will come in a later PR.
There's a TODO in `Struct.cs` to this effect.

File: tools/Google.Cloud.Tools.VersionCompat/Detectors/Struct.cs (hunk 1 above)
Comment (chrisdunelm) [reply tools/Google.Cloud.Tools.VersionCompat/Detectors/Struct.cs]: Done

File: tools/Google.Cloud.Tools.VersionCompat/Detectors/TopLevel.cs (hunk 2 above)
Comment (chrisdunelm) [reply tools/Google.Cloud.Tools.VersionCompat/Detectors/TopLevel.cs]: Fixed.

**Pull request merged**